nmdc_automation/re_iding/scripts/re_id_tool.py

```diff
@@ -92,6 +91,219 @@ def cli(ctx, target):
+        result = db_client["data_object_set"].delete_many({"id": {"$in": has_input}})
+        logging.info(f"Deleted {result.deleted_count} data_object_set records")
+
+        has_output = orphan_metabolomics_activity_record.get("has_output", [])
+        for data_object_id in has_output:
+            updated_record_identifiers.append(("data_object_set", data_object_id, None))
+        result = db_client["data_object_set"].delete_many({"id": {"$in": has_output}})
+        logging.info(f"Deleted {result.deleted_count} data_object_set records")
+
+        logging.info(f"Deleting orphan Metabolomics Activity record: {orphan_metabolomics_activity_record['id']}")
+        updated_record_identifiers.append(
+            ("metabolomics_analysis_activity_set", orphan_metabolomics_activity_record["id"], None)
+            )
+        result = db_client["metabolomics_analysis_activity_set"].delete_one(
+            {"id": orphan_metabolomics_activity_record["id"]}
+            )
+        logging.info(f"Deleted {result.deleted_count} metabolomics_analysis_activity_set records")
+
+
+def _update_metabolomics_input_data_objects(input_data_objects, db_client, api_client, updated_record_identifiers):
+    """
+    Update the input data objects for a Metabolomics OmicsProcessing record.
+    """
+    new_input_data_object_ids = []
+    for input_data_object in input_data_objects:
+        # mint new ID for the data object
+        legacy_id = input_data_object["id"]
+        new_id = api_client.minter("nmdc:DataObject")
+        new_input_data_object_ids.append(new_id)
+        updated_record_identifiers.append(("data_object_set", legacy_id, new_id))
+        # update the data object record
+        do_filter_criteria = {"id": legacy_id}
+        do_update_criteria = {"$set": {"id": new_id, "alternative_identifiers": [legacy_id]}}
+        result = db_client["data_object_set"].update_one(do_filter_criteria, do_update_criteria)
+        logging.info(f"Updated {result.modified_count} data_object_set records")
+    return new_input_data_object_ids
+
+
+def _update_metabolomics_output_data_objects(output_data_objects, metabolomics_new_id, db_client, api_client,
+                                             updated_record_identifiers):
+    """
+    Update the output data objects for a Metabolomics OmicsProcessing record.
+    """
+    new_output_data_object_ids = []
+    for output_data_object in output_data_objects:
+        # mint new ID for the data object
+        legacy_id = output_data_object["id"]
+        new_id = api_client.minter("nmdc:DataObject")
+        new_output_data_object_ids.append(new_id)
+        updated_record_identifiers.append(("data_object_set", legacy_id, new_id))
+        # update the data object record
+        do_filter_criteria = {"id": legacy_id}
+        do_update_criteria = {"$set": {"id": new_id, "alternative_identifiers": [legacy_id], "was_generated_by":
```

Comment:
the data object alternative identifier should be should be old_dms_id, not old data object identifier. this should be the original value for has_output of the omics_processing_set record. see https://github.com/microbiomedata/nmdc-schema/blob/abe2cf1ba29103bdc03d5e8463169e90ff387943/nmdc_schema/metab_id_refactor.py#L601C110-L601C120 which calls the original update_data_products function in Yuri's code.